examples/manual_encrypted_verify.py

```diff
@@ -0,0 +1,199 @@
+
+        except IOError as err:
+            print(f"Couldn't load session from file. Logging in. Error: {err}")
+        except json.JSONDecodeError as err:
+            print(f"Couldn't read JSON file; overwriting")
+
+    if not client.user_id or not client.access_token or not client.device_id:
+        resp = await client.login(ALICE_PASSWORD)
+        if isinstance(resp, LoginResponse):
+            print("Logged in using a password; saving details to disk")
+            write_details_to_disk(resp)
+        else:
+            print(f"Failed to log in: {resp}")
+            sys.exit(1)
+
+    # If your room isn't public, you'll need to invite this user before
+    # it joins the room
+    await client.join(ROOM_ID)
+    await client.sync(full_state=True)
+
+    room = client.rooms[ROOM_ID]
+    print(f"Room {room.name} is encrypted: {room.encrypted}" )
+
+    # nio will transparently encrypt the message for us but other users cannot
+    # read it as we have not shared our keys and haven't received the keys of
+    # anyone else.
+    await client.room_send(
+        room_id=ROOM_ID,
+        message_type="m.room.message",
+        content = {
+            "msgtype": "m.text",
+            "body": "An encrypted message you cannot read"
+        }
+    )
+
+    if (client.should_upload_keys):
+        # We'll upload our public keys for others to use in encrypting messages
+        await client.keys_upload()
+    
+    if (client.should_query_keys):
+        # Since encryption is a two way street, we'll get the public keys
+        # of the other members of our encrypted rooms
+        await client.keys_query()
+
+    # In practice, you want to have a list of previously-known device IDs for
+    # each user you want to trust. Here we require that list as a global var
+    for device_id in BOB_DEVICE_IDS:
+        # When we join an encrypted room and then run sync, we populate our
+        # client.device_store; if we _only_ joined the room or _only_ synced,
+        # client.device_store would be empty.
+
+        # This is the set of known OlmDevices for @bob, a member of an encrypted
+        # room we're in. OlmDevices are devices they've signed in with that we
+        # know about
+        olm_device = client.device_store[BOB_ID][device_id]
+        client.verify_device(olm_device)
+        print(f"Trusting {device_id} for user {BOB_ID}")
+
+    # In this case, we'll trust _all_ of our own devices. NOTE that this is
+    # a SUPER BAD IDEA in practice, but for the purpose of this example
+    # I think it'll be easier since you may end up creating lots of sessions for
+    # @alice as you play with the script
+
+    # Note that client.devices() returns either a DeviceResponse or DeviceError.
+    # In this example, we're implicitly assuming that we get a DeviceResponse
+    # back, but you probably want to catch errors.
+    device_resp = await client.devices()
```

Comment:
Your own devices will be in the device store as well, but I assume you wanted to showcase how getting more info about them is done.

Reply:
Yeah I was thinking that I could sprinkle a bit more about the internals throughout this example but on re-reading I think it's just unclear. Thanks for pointing this out; I'll remove this line and change the loop below.